api/models/site-branch-config.js

```diff
@@ -0,0 +1,106 @@
+const { Op } = require('sequelize');
+const { branchRegex } = require('../utils/validators');
+
+function associate({ SiteBranchConfig, Site }) {
+  // Associations
+  SiteBranchConfig.belongsTo(Site, {
+    foreignKey: 'siteId',
+    allowNull: false,
+  });
+
+  // Scopes
+  SiteBranchConfig.addScope('bySite', id => ({
+    include: [
+      {
+        model: Site,
+        where: { id },
+      },
+    ],
+  }));
+}
+
+function define(sequelize, DataTypes) {
+  const SiteBranchConfig = sequelize.define(
+    'SiteBranchConfig',
+    {
+      branch: {
+        type: DataTypes.STRING,
+        allowNull: true,
+        validate: {
+          is: branchRegex,
+        },
+      },
+      s3Key: {
+        type: DataTypes.STRING,
+        allowNull: true,
+      },
+      config: {
+        type: DataTypes.JSONB,
+        allowNull: true,
+      },
+      context: {
+        type: DataTypes.STRING,
+        allowNull: false,
+        defaultValue: 'preview',
+      },
+    },
+    {
+      tableName: 'site_branch_config',
+      paranoid: true,
+      indexes: [
+        {
+          name: 'site_branch_config_unique_site_id_branch_index',
+          unique: true,
+          fields: ['siteId', 'branch'],
+        },
+      ],
+      validate: {
+        nonPreviewBranchDefined() {
+          if (this.context !== 'preview' && !this.branch) {
+            throw new Error(
+              'Branch attribute cannot be null when context attribute is not preview'
+            );
+          }
+        },
+        nonPreviewS3KeyDefined() {
+          if (this.context !== 'preview' && !this.s3Key) {
+            throw new Error(
+              'The s3Key attribute cannot be null when context attribute is not preview'
```

Comment:
I'm getting this error when trying to edit the live site config on a test site in dev

Reply:
This is probably due to the migration with the updated commits post migration. We may need to run the migration back and forward against the dev database manually.

Reply:
Fixed with update to migration script.